openff/qcsubmit/utils/utils.py

```diff
@@ -1,16 +1,256 @@
+        """
+        Obtain singlepoint records with the specified IDs.
+
+        Records will be returned in the same order as the record ids.
+
+        Parameters
+        ----------
+        record_ids
+            Single ID or sequence/list of records to obtain
+        missing_ok
+            Unlike a ``PortalClient``, this argument is ignored. If set to
+            True, a warning will be printed. Any missing records will cause a
+            ``RuntimeError`` to be raised.
+        include
+            Additional fields to include in the returned record
+
+        Returns
+        -------
+        :
+            If a single ID was specified, returns just that record. Otherwise,
+            returns a list of records.
+        """
+        if missing_ok:
+            logger.warning("missing_ok provided but unused by CachedPortalClient")
+        if unpack := not isinstance(record_ids, Sequence):
+            record_ids = [record_ids]
+        res = get_records_with_cache(
+            client=self,
+            record_cache=self.record_cache,
+            record_type=SinglepointRecord,
+            record_ids=record_ids,
+            include=include,
+            force_fetch=False,
+        )
+        if unpack:
+            return res[0]
+        else:
+            return res
+
+    def get_torsiondrives(
+        self,
+        record_ids: Union[int, Sequence[int]],
+        missing_ok: bool = False,
+        *,
+        include: Optional[Iterable[str]] = None,
+    ) -> Union[Optional[TorsiondriveRecord], List[Optional[TorsiondriveRecord]]]:
+        """
+        Obtain torsiondrive records with the specified IDs.
+
+        Records will be returned in the same order as the record ids.
+
+        Parameters
+        ----------
+        record_ids
+            Single ID or sequence/list of records to obtain
+        missing_ok
+            Unlike a ``PortalClient``, this argument is ignored. If set to
+            True, a warning will be printed. Any missing records will cause a
+            ``RuntimeError`` to be raised.
+        include
+            Additional fields to include in the returned record
+
+        Returns
+        -------
+        :
+            If a single ID was specified, returns just that record. Otherwise,
+            returns a list of records.
+        """
+        if missing_ok:
+            logger.warning("missing_ok provided but unused by CachedPortalClient")
+        if unpack := not isinstance(record_ids, Sequence):
+            record_ids = [record_ids]
+        res = get_records_with_cache(
+            client=self,
+            record_cache=self.record_cache,
+            record_type=TorsiondriveRecord,
+            record_ids=record_ids,
+            include=include,
+            force_fetch=False,
+        )
+        if unpack:
+            return res[0]
+        else:
+            return res
+
+    @contextmanager
+    def _no_session(self):
```

Comment:
(not blocking) this looks like a good addition. If this is strictly better than `no_internet`, then I'd be in favor of replacing all usages of the latter in a subsequent PR.

Reply:
I think you're right about this too. I didn't know it at the time, but `test_manager` (the only usage of `no_internet`) only looks like it works because it's reusing the client with the lambda (thus preserving the session), not because it's reading from the cache. In fact, setting `client._req_session = None` in that test causes it to fail because the dataset fetching cache and the cache needed by the `to_records` call are separate. Now that `portal_client_manager` is used in many other tests, I think it would be okay to delete the `test_manager.py` file with its single test and the `no_internet` manager entirely. 

Do you want me to do that here, or in a separate PR as you mentioned?

Reply:
Good catch about the test not doing what we thought. Let's make those changes in a separate PR - This one has so much scope that it's getting tough to review, and it'll be good to consolidate our gains.